website/src/types/Task.ts

```diff
@@ -29,12 +29,26 @@ export interface BaseTask {
   type: TaskType;
 }
 
-export interface TaskResponse<Task extends BaseTask> {
+export interface TaskAvailableResponse<Task extends BaseTask> {
   id: string;
   userId: string;
   task: Task;
 }
 
+interface TaskAvailable<Task extends BaseTask> extends TaskAvailableResponse<Task> {
+  status: "AVAILABLE";
+}
+
+interface AwaitingInitialTask {
+  status: "AWAITING_INITIAL";
+}
+
+interface NoTaskAvailable {
+  status: "NONE_AVAILABLE";
+}
```

Comment:
these could be an enum

Reply:
Are you referring to the values for `status`? Sure I can do that, but can you elaborate on what additional benefit we get from replacing the strings with enum values? The `TaskResponse` is already a tagged union and as far as I know (which isn't much, I'm new to typescript) the type system already checks that the strings match, that we have handled all cases in a switch etc, and the autocomplete (at least for me) completes them. Just looking to understand it better and learn more.

Reply:
Yes I am referring to `status`, the enum could also be a string, that's ok, an enum helps show intent better than 3 different interfaces (and another type that combines these interfaces), also reduces the amount of code you need to write, and most importantly, understand.
```ts
export enum TaskAvailability {
   AVAILABLE = "AVAILABLE",
   AWAITING_INITIAL = "AWAITING_INITIAL",
   NONE_AVAILABLE = "NONE_AVAILABLE"
}
```

Reply:
The reason we can't use an enum instead of the interfaces is that the TaskAvailable one has data (via TaskAvailableResponse), unless of course there is a way to do this with enums, I'm still new to typescript.

Reply:
that's a point against combining the status with the task, a task being available or awaiting first load has nothing to do with which task nor its contents.

Reply:
I disagree, the task is what we are loading, it being available and having it are the same thing.

Anyway, now that we have discussed it a bunch and I think we are on the same page, what changes would you like me to make.

Reply:
Yeah it dragged along a bit too much, you can just leave it like this for now.